### ISSUE TITLE
bug: CallbackSignatureShouldMatchMockedMethodAnalyzer assumes parameters match on validation failure

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#995]
## Summary

The callback signature validation in `CallbackSignatureShouldMatchMockedMethodAnalyzer` returns `true` ("parameters match") when type resolution fails, causing false negatives.

## Details

**File:** `src/Analyzers/CallbackSignatureShouldMatchMockedMethodAnalyzer.cs`, lines 210-221

```csharp
TypeSyntax? lambdaParameterTypeSyntax = lambdaParameter.Type;
if (lambdaParameterTypeSyntax is null)
{
    return true; // Can't validate, assume ok
}

TypeInfo lambdaParameterType = semanticModel.GetTypeInfo(lambdaParameterTypeSyntax, cancellationToken);
ITypeSymbol? lambdaParameterTypeSymbol = lambdaParameterType.Type;

if (lambdaParameterTypeSymbol is null)
{
    return true; // Can't validate, assume ok
}
```

When type resolution fails (compilation errors, unresolved types), the method returns `true` meaning "parameters match." This silently suppresses the Moq1100 diagnostic.

## Impact

Compilation errors or unresolved types cause false negatives. Developers with incorrect callback signatures get no diagnostic when there are also unresolvable types in the file.

## Discovered by

Automated codebase investigation (Silent Failure Hunter agent).

**Comment by rjmurillo** [comment]
## Code Review: Issue #995

**Reviewed:** `src/Analyzers/CallbackSignatureShouldMatchMockedMethodAnalyzer.cs`, lines 178-231

---

### Verdict: Confirmed Bug

The issue report is accurate. The two `return true` paths in `ParameterTypesMatch` are incorrect. Here is the precise execution path that produces the false negative.

**Call chain:**

```
ValidateParameters
  foreach (IMethodSymbol mockedMethod in mockedMethodSymbols)
    if (ParametersMatch(...))   // returns true on first "assume ok" hit
      return;                  // diagnostic is suppressed
```

`ParametersMatch` iterates each parameter pair and calls `ParameterTypesMatch`. The moment `ParameterTypesMatch` returns `true`, `ParametersMatch` returns `true`, and `ValidateParameters` exits without reporting. One unresolvable type in the parameter list is sufficient to suppress the entire diagnostic, regardless of whether the other parameters are wrong.

---

### Case 1: `lambdaParameter.Type is null` (line 210-213)

```csharp
TypeSyntax? lambdaParameterTypeSyntax = lambdaParameter.Type;
if (lambdaParameterTypeSyntax is null)
{
    return true; // Can't validate, assume ok
}
```

`ParameterSyntax.Type` is `null` only when the lambda parameter has no type annotation, i.e., an implicitly typed parameter like `(x) => ...`. In that case the compiler infers the type. The semantic model can still resolve the inferred type via `GetTypeInfo` on the parameter itself (not on `lambdaParameter.Type`). The current code skips that resolution entirely and assumes a match.

The traycer comment proposes `return false` here. That is the correct direction but needs nuance: if the type is implicitly typed and can be resolved, a false positive is introduced. The safest minimal fix is to attempt resolution via `semanticModel.GetTypeInfo(lambdaParameter, cancellationToken)` and fall back to `return false` only if that also fails.

---

### Case 2: `lambdaParameterTypeSymbol is null` (line 218-221)

```csharp
ITypeSymbol? lambdaParameterTypeSymbol = lambdaParameterType.Type;

if (lambdaParameterTypeSymbol is null)
{
    return true; // Can't validate, assume ok
}
```

This fires when the type annotation is present but unresolvable (compilation error, missing reference, unknown type). Returning `true` here is the core of the reported bug. There is a real type annotation that the compiler cannot match, and the analyzer concludes "looks fine." Changing this to `return false` is correct. The parameter type is written, it cannot be verified, and that is not a passing state.

---

### Impact on `ValidateParameters`

The `mockedMethodSymbols` sequence is populated by `GetAllMatchingMockedMethodSymbolsFromSetupMethodInvocation`, which returns an empty list when setup lambda resolution fails. If `mockedMethodSymbols` is empty, the `foreach` loop does not execute, and the diagnostic fires correctly (assuming `lambdaParameters.Count > 0`). The false negative only occurs when at least one method symbol is resolved and `ParametersMatch` returns `true` due to the bailout.

This means the bug is scoped to cases where the mocked method itself resolves (normal compilation) but a lambda parameter type does not (e.g., a typo in the type name). That is exactly the scenario described in the issue.

---

### Assessment of the Proposed Fix (traycer comment)

The traycer plan is directionally correct. Changing both bailouts to `return false` eliminates the false negatives. Two concerns:

1. **Case 1 false positive risk.** Implicitly typed lambda parameters are valid C# and commonly written. If a developer writes `.Callback((x) => { })` against a matching method, case 1 fires and returns `false`, which causes `ParametersMatch` to return `false`, and the diagnostic fires incorrectly. The fix must handle the implicit-type case separately before defaulting to `false`.

2. **Test coverage gap.** The proposed test only covers the unresolvable explicit type case. A test for implicitly typed parameters with a matching method should confirm no diagnostic fires after the fix.

---

### Recommended Fix

In `ParameterTypesMatch`:

- Case 1 (`lambdaParameterTypeSyntax is null`): attempt `semanticModel.GetTypeInfo(lambdaParameter, cancellationToken).Type` to resolve the inferred type. If still null, return `false`. If resolved, continue with the existing conversion check against `mockedParameterType`.
- Case 2 (`lambdaParameterTypeSymbol is null`): change `return true` to `return false`. No additional handling needed.

The comment text should change from "Can't validate, assume ok" to "Type is unresolvable; treat as mismatch."

---

### Summary

| Location | Current | Correct |
|---|---|---|
| Line 212 | `return true` | Attempt inferred type resolution; `return false` if still null |
| Line 220 | `return true` | `return false` |

The traycer plan is a reasonable starting point. Validate the implicit-type case before merging.
